omegaconf/listconfig.py

```diff
@@ -390,13 +390,21 @@ def _get_node(
             if value is not None:
                 if isinstance(key, slice):
                     assert isinstance(value, list)
-                    for v in value:
+                    for idx, v in enumerate(value):
                         if throw_on_missing_value and v._is_missing():
-                            raise MissingMandatoryValue("Missing mandatory value")
+                            self._format_and_raise(
+                                key=idx,
+                                value=None,
+                                cause=MissingMandatoryValue("Missing mandatory value"),
+                            )
                 else:
                     assert isinstance(value, Node)
                     if throw_on_missing_value and value._is_missing():
-                        raise MissingMandatoryValue("Missing mandatory value")
+                        self._format_and_raise(
+                            key=key,
+                            value=None,
+                            cause=MissingMandatoryValue("Missing mandatory value"),
+                        )
```

Comment:
Nice, but add corresponding tests of _get_node that confirms that this is working (especially the slice case).
Feel free to improve test_get_node_throw_on_missing_value to do that instead of making test_errors even bigger.

Reply:
By adding tests (https://github.com/omry/omegaconf/pull/503/commits/7d7026fb8cc8463e0a5c291abdca6a6549176a25) I discovered that there was an error in the slice case. I have seen the light. Test everything.